### PR TITLE
Added support for `fail-if-unchanged`

### LINKED ISSUE #394: [Feature] Fail if unchanged
### Is this feature missing in the latest version?

- [X] I'm using the latest release

### Is your feature request related to a problem? Please describe.

Similar to https://github.com/tj-actions/verify-changed-files/issues/320.

I currently need to use an extra step to fail if files are unchanged which would be nice cold be handled by this action itself via an input.

### Describe the solution you'd like?

An input like fail-if-unchanged that makes the action fail if it detects no changes.

image

### Describe alternatives you've considered?

My current workaround using a separate step that always fails and only run that if this action detected no changes.

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Larspolo**
Fixed #394

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -43,16 +43,27 @@ jobs:
         run: |
           echo "Changed files (Not expected): ${{ steps.changed_files_not_expected.outputs.changed_files }}"
           exit 1
-      - name: Test dont fail if not changed
+      - name: Test dont fail if not changed and fail-if-changed is true
```
Comment (jackton1) [inline .github/workflows/test.yml:46]: Can we revert the changes to this line.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -43,16 +43,27 @@ jobs:
...
         uses: ./
-        id: changed_files_not_expected_fail
+        id: unchanged_files_not_expected_fail
```
Comment (jackton1) [inline .github/workflows/test.yml:48]: Same as above

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -88,7 +99,7 @@ jobs:
         run: |
           echo "No Changes found or missing changed files: (Not expected)"
           exit 1
-      - name: Test fail if changed
+      - name: Test fail if changed and fail-if-changed is true
```
Comment (jackton1) [inline .github/workflows/test.yml:102]: Same as above

File: README.md [hunk 4]
```diff
@@ -132,8 +132,13 @@ Support this project with a :star:
     # Default: "false"
     fail-if-changed: ''
 
-    # Message to display when files have changed and the 
-    # `fail-if-changed` input is set to `true`. 
+    # Indicates whether to fail if no files have changed.
```
Comment (jackton1) [inline README.md:135]: The changes in this file can be reverted since it’s automatically generated

File: action.yml [hunk 5]
```diff
@@ -21,8 +21,12 @@ inputs:
     description: 'Indicates whether to fail if files have changed.'
     default: "false"
     required: false
+  fail-if-unchanged:
+    description: 'Indicates whether to fail if no files have changed.'
+    default: "false"
+    required: false
   fail-message:
-    description: 'Message to display when files have changed and the `fail-if-changed` input is set to `true`.'
+    description: 'Message to display when `fail-if-changed` or `fail-if-unchanged` is set to `true` and triggered.'
```
Comment (jackton1) [inline action.yml:29]:
```suggestion
    description: 'Message to display when `fail-if-changed` or `fail-if-unchanged` is set to `true`.'
```

**Review by jackton1 — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (Larspolo) [reply .github/workflows/test.yml]: Sure, reverted the requested changes

File: README.md (hunk 4 above)
Comment (Larspolo) [reply README.md]: Missed that, reverted it

File: action.yml (hunk 5 above)
Comment (Larspolo) [reply action.yml]: Changed to suggestion

**Comment by jackton1** [comment]
@all-contributors please add @Larspolo for code, test and doc

**Comment by jackton1** [comment]
Nice work on this @Larspolo.

Sorry for the late review of the PR, the change should be available in the next minor release.
